gst_msdk/gst-libs/mfx/gstmfxfilter.c

```diff
@@ -888,14 +888,20 @@ gst_mfx_filter_set_deinterlace_mode (GstMfxFilter * filter,
   mfxExtVPPDeinterlacing *ext_deinterlacing;
 
   g_return_val_if_fail (filter != NULL, FALSE);
+
+#if MSDK_CHECK_VERSION(1,19)
```

Comment:
It does mean the same thing as the original code, I don't see in which case it wouldn't?
If you define G_DISABLE_CHECKS, the whole block gets removed (so does the error, but that doesn't fix it :))

I'd rather stay with g_return_val_if_fail since it's used earlier in this file too. It does a bit more than just a return, I've looked into it and it gives compiler hint and does logging.